linkcheck/director/aggregator.py

```diff
@@ -87,16 +87,22 @@ def visit_loginurl(self):
         session = requests.Session()
         # XXX user-agent header
         # XXX timeout
+        log.debug(LOG_CHECK, "Getting login form %s", url)
         response = session.get(url)
+        response.raise_for_status()
         cgiuser = self.config["loginuserfield"]
         cgipassword = self.config["loginpasswordfield"]
         form = loginformsearch.search_form(response.text, cgiuser, cgipassword)
+        if not form:
+            raise LinkCheckerError("Login form not found at %s" % url)
         form.data[cgiuser] = user
         form.data[cgipassword] = password
         for key, value in self.config["loginextrafields"].items():
             form.data[key] = value
         formurl = parse.urljoin(url, form.url)
+        log.debug(LOG_CHECK, "Posting login data to %s", formurl)
         response = session.post(formurl, data=form.data)
+        response.raise_for_status()
```

Comment:
does someone catch those exceptions above?

Reply:
Yes (and reraises):

https://github.com/linkchecker/linkchecker/blob/0f18c9b8f0784dbb0acf2d0867081df52d48bb36/linkcheck/director/__init__.py#L36-L41

It is an attempt reduce user confusion again, to give the best information about why the login failed.